diff --git a/eng/Versions.props b/eng/Versions.props
index 605321faf..bd747b098 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -42,10 +42,10 @@
     <!-- Roslyn Testing -->
     <MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.21126.1</MicrosoftCodeAnalysisTestingVersion>
     <!-- Libs -->
-    <SystemCollectionsImmutableVersion>1.3.1</SystemCollectionsImmutableVersion>
-    <SystemComponentModelCompositionVersion>4.7.0</SystemComponentModelCompositionVersion>
-    <SystemDirectoryServicesVersion>4.7.0</SystemDirectoryServicesVersion>
-    <SystemReflectionMetadataVersion>1.4.2</SystemReflectionMetadataVersion>
+    <SystemCollectionsImmutableVersion>5.0.0</SystemCollectionsImmutableVersion>
+    <SystemComponentModelCompositionVersion>5.0.0</SystemComponentModelCompositionVersion>
+    <SystemDirectoryServicesVersion>5.0.0</SystemDirectoryServicesVersion>
+    <SystemReflectionMetadataVersion>5.0.0</SystemReflectionMetadataVersion>
     <MicrosoftVisualBasicVersion>10.1.0</MicrosoftVisualBasicVersion>
     <MicrosoftBuildLocatorVersion>1.1.2</MicrosoftBuildLocatorVersion>
     <SQLitePCLRawVersion>1.1.2</SQLitePCLRawVersion>
